Intall error on Cori

I am trying to build Adios2 on Cori but getting the following error. Building was ok but installation was failed:

```
-- Installing: /project/projectdirs/m499/Software/test/adios2/v2.1.0-2902-g83da09c8/cori_haswell/intel/lib64/libadios2_sst.a
-- Installing: /project/projectdirs/m499/Software/test/adios2/v2.1.0-2902-g83da09c8/cori_haswell/intel/bin/sst_conn_tool
CMake Error at source/adios2/toolkit/sst/util/cmake_install.cmake:45 (file):
  file RPATH_CHANGE could not write new RPATH:

    $ORIGIN/../lib64

  to the file:

    /project/projectdirs/m499/Software/test/adios2/v2.1.0-2902-g83da09c8/cori_haswell/intel/bin/sst_conn_tool

  No valid ELF RPATH or RUNPATH entry exists in the file;
Call Stack (most recent call first):
  source/adios2/toolkit/sst/cmake_install.cmake:45 (include)
  source/adios2/cmake_install.cmake:54 (include)
  source/cmake_install.cmake:37 (include)
  cmake_install.cmake:43 (include)


Makefile:73: recipe for target 'install' failed
```

Here is my cmake command:
```
cmake 
-DCMAKE_INSTALL_PREFIX=/dir/to/install 
-DCMAKE_BUILD_TYPE=RelWithDebInfo 
-DADIOS2_USE_MPI=ON 
-DADIOS2_USE_Fortran=ON 
-DADIOS2_USE_Python=OFF 
-DADIOS2_BUILD_SHARED_LIBS=ON 
-DADIOS2_BUILD_EXAMPLES_EXPERIMENTAL=OFF 
-DADIOS2_BUILD_TESTING=OFF 
-DADIOS2_USE_ADIOS1=OFF 
-DADIOS2_USE_HDF5=OFF 
-DADIOS2_USE_SST=ON 
..
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,11 +25,6 @@ set(CMAKE_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/adios2
   CACHE STRING "Installation CMake subdirectory")
 mark_as_advanced(CMAKE_INSTALL_CMAKEDIR)
 
-# On Mac OS X, set the dir. included as part of the installed library's path:
-if (BUILD_SHARED_LIBS AND NOT DEFINED CMAKE_INSTALL_NAME_DIR)
-  set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-endif ()
-
 list(INSERT CMAKE_MODULE_PATH 0 "${PROJECT_SOURCE_DIR}/cmake")
 if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
   set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY
@@ -46,15 +41,6 @@ endif()
 
 # Let windows builds auto-export dll symbols
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
-
-#------------------------------------------------------------------------------#
-# RPATH defaults
-#------------------------------------------------------------------------------#
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  string(REGEX REPLACE "[^/]+" ".." relative_base "${CMAKE_INSTALL_LIBDIR}")
-  set(CMAKE_INSTALL_RPATH "$ORIGIN/${relative_base}/${CMAKE_INSTALL_LIBDIR}")
-endif()
 
 #------------------------------------------------------------------------------#
 # Silence MSVC warnings
@@ -146,11 +132,6 @@ if(ADIOS2_HAVE_MPI)
   add_definitions(-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX)
 endif()
 
-# Various homebrew MPI fortran installations break rpath usage on OSX
-if(APPLE AND ADIOS2_HAVE_Fortran AND ADIOS2_HAVE_MPI)
-  set(CMAKE_MACOSX_RPATH OFF)
-endif()
-
 set(ADIOS2_CONFIG_OPTS
     BZip2 ZFP SZ MGARD MPI DataMan SSC SST ZeroMQ HDF5 Python Fortran SysVShMem Profiling Endian_Reverse
 )
@@ -164,6 +145,31 @@ configure_file(
 install(FILES ${PROJECT_BINARY_DIR}/source/adios2/ADIOSConfig.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/adios2
 )
+
+#------------------------------------------------------------------------------#
+# Some specialized shared library and RPATH handling
+#------------------------------------------------------------------------------#
+if(BUILD_SHARED_LIBS)
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+  if(APPLE)
+    # On Mac OS X, set the dir included as part of the installed library's path:
+    if(NOT DEFINED CMAKE_INSTALL_NAME_DIR)
+      set(CMAKE_INSTALL_NAME_DIR
+        "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+    endif ()
+
+    # Various homebrew MPI fortran installations break rpath usage on OSX
+    if(APPLE AND ADIOS2_HAVE_Fortran AND ADIOS2_HAVE_MPI)
+      set(CMAKE_MACOSX_RPATH OFF)
+    endif()
+
+  elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    # Linux needs some specialized RPATH handling
+    string(REGEX REPLACE "[^/]+" ".." relative_base "${CMAKE_INSTALL_LIBDIR}")
+    set(CMAKE_INSTALL_RPATH "$ORIGIN/${relative_base}/${CMAKE_INSTALL_LIBDIR}")
+  endif()
+endif()
 
 #------------------------------------------------------------------------------#
 # Third party libraries

